### ISSUE TITLE
cannot upgrade an HA cluster with external etcd from 1.10.4 to 1.11.0-beta.2

### ISSUE THREAD

**Issue by chuckha** [kubernetes/kubeadm#936]
## Is this a BUG REPORT or FEATURE REQUEST?

BUG REPORT

## Versions
1.10.4 => 1.11.0-beta.2(ish)

**kubeadm version** (use `kubeadm version`):
kubeadm version: &version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-beta.2.57+8bb1309efed13d-dirty", GitCommit:"8bb1309efed13d572c12bf29397e533e07579b56", GitTreeState:"dirty", BuildDate:"2018-06-12T19:00:36Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}

## What happened?
I ran `kubeadm upgrade apply v1.11.0-beta.2 --allow-experimental-upgrades`  and got a failure 

```
[upgrade/apply] FATAL: failed to upgrade the etcd CA certificate and key: failure loading etcd/ca certificate: couldn't load the private key file /etc/kubernetes/pki/etcd/ca.key: open /etc/kubernetes/pki/etcd/ca.key: no such file or directory    
```

So I ran `kubeadm upgrade apply v1.11.0-beta.2 --allow-experimental-upgrades --etcd-upgrade=false`

and got the exact same error:

```
root@ip-10-0-0-7:~# kubeadm upgrade apply v1.11.0-beta.2 --allow-experimental-upgrades --etcd-upgrade=false                                                                                                                                           
[preflight] Running pre-flight checks.
[upgrade] Making sure the cluster is healthy:
[upgrade/config] Making sure the configuration is correct:
[upgrade/config] Reading configuration from the cluster...
[upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
I0618 20:11:41.262603   26371 feature_gate.go:230] feature gates: &{map[]}
[upgrade/version] You have chosen to change the cluster version to "v1.11.0-beta.2"
[upgrade/versions] Cluster version: v1.10.4
[upgrade/versions] kubeadm version: v1.11.0-beta.2.57+8bb1309efed13d-dirty
[upgrade/confirm] Are you sure you want to proceed with the upgrade? [y/N]: y
[upgrade/prepull] Will prepull images for components [kube-apiserver kube-controller-manager kube-scheduler etcd]
[upgrade/apply] Upgrading your Static Pod-hosted control plane to version "v1.11.0-beta.2"...
Static pod: kube-apiserver-ip-10-0-0-9.us-west-2.compute.internal hash: 4d7e14fe5e8fff6f1afc4505c7fc6719
Static pod: kube-controller-manager-ip-10-0-0-9.us-west-2.compute.internal hash: 10c1dc0406aa60f0d7edab39d957c7a3
Static pod: kube-scheduler-ip-10-0-0-9.us-west-2.compute.internal hash: a45cd09c7a02c8a7e64580248df4b50d
[upgrade/staticpods] Writing new Static Pod manifests to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests530298575"
[controlplane] wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests530298575/kube-apiserver.yaml"                                                                                                
[controlplane] wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests530298575/kube-controller-manager.yaml"                                                                              
[controlplane] wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests530298575/kube-scheduler.yaml"                                                                                                
[upgrade/apply] FATAL: failed to upgrade the etcd CA certificate and key: failure loading etcd/ca certificate: couldn't load the private key file /etc/kubernetes/pki/etcd/ca.key: open /etc/kubernetes/pki/etcd/ca.key: no such file or directory    
```

## What you expected to happen?
I expected the second command to not need ca.key for etcd.

## How to reproduce it (as minimally and precisely as possible)?
Upgrade a cluster without an etcd/ca.key file.

**Comment by chuckha** [comment]
I tried again specifying my external etcd config file and got the same result.

**Comment by chuckha** [comment]
I traced it to this code

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/upgrade/staticpods.go#L184

I need to think about why we're generating etcd certs during the kube-apiserver upgrade.

/cc @stealthybox

**Comment by timothysc** [comment]
/assign @liztio 

I'm not certain this still is an issue anymore.

**Comment by timothysc** [comment]
@liz I'd use this issue to verify upgrade process.

**Comment by liztio** [comment]
```
[controlplane] wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests991474213/kube-apiserver.yaml"
[controlplane] wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests991474213/kube-controller-manager.yaml"
[controlplane] wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/tmp/kubeadm-upgraded-manifests991474213/kube-scheduler.yaml"
[certificates] Using the existing etcd/ca certificate and key.
[upgrade/staticpods] Moved new manifest to "/etc/kubernetes/manifests/kube-apiserver.yaml" and backed up old manifest to "/etc/kubernetes/tmp/kubeadm-backup-manifests-2018-09-07-17-23-39/kube-apiserver.yaml"
[upgrade/staticpods] Waiting for the kubelet to restart the component
```

Looks like this is fixed in v1.12
